docs/migration-to-NRI.md

```diff
@@ -0,0 +1,182 @@
+    $ sudo vim /.../kubeadm-flags.env
+    KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock --pod-infra-container-image=registry.k8s.io/pause:3.9"
+
+    $ sudo vim /etc/sysconfig/kubelet 
+    KUBELET_EXTRA_ARGS= --container-runtime-endpoint=/var/run/containerd/containerd.sock <- Remember this aswell
+
+    $ systemctl restart containerd
+    $ systemctl restart kubelet
+    ```
+
+- Edit the containerd config file and look for the section [plugins."io.containerd.nri.v1.nri"] and replace "disable = true" with "disable = false":
+  ```console
+  $ vim /etc/containerd/config.toml
+  ```
+  ```toml
+  [plugins."io.containerd.nri.v1.nri"]
+    disable = false
+    disable_connections = false
+    plugin_config_path = "/etc/nri/conf.d"
+    plugin_path = "/opt/nri/plugins"
+    plugin_registration_timeout = "5s"
+    plugin_request_timeout = "2s"
+    socket_path = "/var/run/nri/nri.sock"
+  ```
+
+## CRI-O
+- Replace the crio version in the system with 1.26.2 or newer.
+
+- Same thing with the Kubelet runtime endpoint, but change it to the crio socket and then restart the crio service.
+
+- Edit the crio config and look for the "[crio.nri]" section, replace "enable_nri = false" with "enable_nri = true" and uncomment the NRI config lines:
+  ```console
+  $ vim /etc/crio/crio.conf
+  ```
+
+  ```
+  # CRI-O NRI configuration.
+  [crio.nri]
+
+  # Globally enable or disable NRI.
+  enable_nri = true
+
+  # NRI configuration file to use.
+  nri_config_file = "/etc/nri/nri.conf"
+
+  # NRI socket to listen on.
+  nri_listen = "/var/run/nri/nri.sock"
+
+  # NRI plugin directory to use.
+  nri_plugin_dir = "/opt/nri/plugins"
+  ```
+
+## Build the policies
+  ```console
+    $ git clone https://github.com/jukkar/nri-resmgr.git
+    $ cd nri-resmgr
+    # Switch to a stable branch
+    $ git checkout preview
+    $ make
+    $ make images
+  ```
+
+## Apply required crds
+```console
+kubectl apply -f deployment/base/crds/noderesourcetopology_crd.yaml
+```
+
+## Import the image of the NRI plugin you want to run
+### Containerd
+  ```console
+  $ ctr -n k8s.io images import build/images/nri-resmgr-topology-aware-image-9797e8de7107.tar
+  ```
+
+### CRI-O
+- See the section [below](#steps-for-a-more-real-life-migration-using-self-hosted-image-repository) for instructions on how to push the images to a registry, then pull from there.
+
+## Then deploy the plugin
+  ```console
+  $ kubectl apply -f build/images/nri-resmgr-topology-aware-deployment.yaml
+  ```
+
+## Deploy your test pod
+
+## See the resources the pod got assigned with
+  ```console
+  $ kubectl exec $pod -c $container  -- grep allowed_list: /proc/self/status
+  # Output should look similar to the output of CRI-RM
+  ```
+
+# Steps for a more real-life migration using self-hosted image repository
+- ### Same steps as above for enabling NRI with Containerd/CRI-O and building the images.
+- ### Load the .tar file into your image repository:
```

Comment:
When markdown-rendered to HTML these look weird, probably because they are both list items and <H3>-headers. I suspect you meant them to be either but not both.